anvil/src/eth/backend/executor.rs

```diff
@@ -258,11 +267,31 @@ impl<'a, 'b, DB: Db + ?Sized, Validator: TransactionValidator> Iterator
 
         trace!(target: "backend", "[{:?}] executing", transaction.hash());
         // transact and commit the transaction
-        let ExecutionResult { exit_reason, out, gas_used, logs, .. } =
-            evm.inspect_commit(&mut inspector);
+        let exec_result = match evm.inspect_commit(&mut inspector) {
+            Ok(exec_result) => exec_result,
+            Err(err) => {
+                warn!(target: "backend", "[{:?}] failed to execute: {:?}", transaction.hash(), err);
+                return Some(TransactionExecutionOutcome::DatabaseError(
+                    transaction,
+                    DatabaseError::TransactionNotFound(H256::from_str("0x").unwrap()),
```

Comment:
Note that this error is very much a placeholder—We probably need another variant to properly represent the inspector failing.